README.md

```diff
@@ -49,10 +49,147 @@ $ yarn lint
+- use-case already have recording → records new traffic and compares it to the old one →
+    - if it matches → do not save recording
+    - if it doesn’t match → save new recording next to old one with suffix `-new`
+
+### Test with new mocked traffic
+
+```shell
+$ yarn test:with_new_traffic
+```
+
+This command run tests without live traffic, similar as `yarn test`, but uses new recordings instead of default ones (if present). New recordings are always newly recorded traffic that does not match the default old one and are located next to the old one with suffix `-new`.
+
+### Test and replace old recording with new one
+
+**⚠️ This command should be run only when you’re sure that tests are passing with new recordings → `yarn test:with_new_traffic`**
+
+```shell
+$ yarn test:update_traffic
+```
+
+This commands run tests with mocked traffic, but also replaces old recording with new one (if present) before loading recordings - this means that tests are run with mocked new traffic already.
+
+### Test with live traffic with development reporting
+
+```shell
+$ yarn test:record:dev
+```
+
+This command run tests with live traffic and record it, similar as `yarn test:record`. The script implementation looks like this:
+
+```shell
+$ TEST_ENV='dev' yarn test:record --group=live/safe
+```
+
+`TEST_ENV` sets development environment for reporter - to use local reporter that just logs the provider changes into console
+
+`--group=live/safe` filters what tests to run, thanks to [jest-runner-groups](https://www.npmjs.com/package/jest-runner-groups) 
+
+### Test with live traffic with production reporting
+
+```shell
+$ yarn test:record:prod
+```
+
+This command run tests with live traffic and record it, similar as `yarn test:record` or `yarn test:record:dev`. Only difference between `test:record:dev` is that it uses production reporter that reports provider changes to the public slack channel.
+
+Slack channel can be changed with env variable `PROD_REPORTING_DESTINATION`. It expects channel id.
+
+### Test and update snapshots
+
+Since lot of tests written in station uses jest snapshots for comparing values from map, you might need to update snapshots to modify/add/remove some of them. You can do this simply by adding option `-u` or `--updateSnapshot`. For example:
+
+```shell
+$ yarn test -u
+
+$ yarn test:record --updateSnapshot
+```
+
+---
+
+### Overall filtering of tests
+
+In each of command described above, you can use argument to specify integration, for example:
+
+```shell
+$ yarn test chat/messages/maps/slack
+
+$ yarn test:record chat/messages/maps/slack
+
+$ TEST_ENV='dev' yarn test:record chat/messages
+```
+
+This is important mainly in case of testing with live traffic when new recordings are stored and compared with old ones for each test that you run. For example if you want to record new traffic just for slack provider in profile `chat/messages`, you don’t want to record new traffic also for other providers in this profile.
```

Comment:
Move this paragraph above the code block above, to continue with the explained example.

```suggestion
This is necessary whenever you record a traffic to limit interaction with live APIs only to the changed maps and tests.

For example, if you want to record new traffic just for the `slack` provider and profile `chat/messages`, and ignore other providers:
```